### PR TITLE
[git]: Add type and fallback tests for it

### LINKED ISSUE #278: [git] Add git (proxy) type
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
Currently we don't support a git type.

**What you expected to happen**:
Support git proxying.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
We have to proxy unfortunately as git doesn't allow redirects without user interaction so we have to proxy.

Additionally we should first support https based proxying only. This should already cover most of the "consumer" facing vanity urls. ssh based proxying could be looked into in the long term.

Code inspiration: https://github.com/niemeyer/gopkg

Sidenote: Should probably work with type=path as well.

Simple case:
```
git.okkur.org                                  86400 IN A   127.0.0.1
_redirect.torproxy.git.okkur.org                  30 IN TXT     "v=txtv0;to=https://github.com/okkur/torproxy.git;type=git"
```

Path case:
```
git.okkur.org                                  86400 IN A   127.0.0.1
_redirect.git.okkur.org                           30 IN TXT     "v=txtv0;to=https://about.okkur.org/;type=path"
_redirect.torproxy.git.okkur.org                  30 IN TXT     "v=txtv0;to=https://github.com/okkur/torproxy.git;type=git"
```

### PR DESCRIPTION
**Pull request by erbesharat**
**What this PR does / why we need it**:
Adds `git` type and fallback tests for it

**Which issue this PR fixes**:
fixes #278

### CODE REVIEW

File: path.go [hunk 1]
```diff
@@ -87,7 +87,14 @@ func (p *Path) RedirectRoot() error {
 // It will use custom regex to parse the path if it's provided in
 // the given record.
 func zoneFromPath(r *http.Request, rec record) (string, int, []string, error) {
-	path := fmt.Sprintf("%s?%s", r.URL.Path, r.URL.RawQuery)
+	path := r.URL.Path
+
+	// Check if request is from a Git client
+	if strings.HasPrefix(r.Header.Get("User-Agent"), "git") {
+		path = "/" + strings.Split(r.URL.Path, "/")[1]
```
Comment (stp-ip) [inline path.go]: What is your reasoning behind this?

**Comment by stp-ip** [comment]
Looking forward to pushing this live and use it for syna-starter as first use-case \o/

File: path.go (hunk 1 above)
Comment (erbesharat) [reply path.go]: Git sends the request to a path like `/torproxy/info/refs?service=git-upload-pack` and if we don't split the first part of the path in here it would lead to this:
```
2019-09-30T22:19:24.772+03:30 [INFO] 127.0.0.1:57746 - 62342 "TXT IN _redirect.git.example.test. udp 44 false 512" NOERROR qr,aa,rd 207 0.000153389s
2019-09-30T22:19:24.773+03:30 [INFO] 127.0.0.1:35984 - 37431 "TXT IN _redirect.refs.info.torproxy.git.example.test. udp 63 false 512" NXDOMAIN qr,aa,rd 144 0.000183976s
2019-09-30T22:19:24.773+03:30 [INFO] 127.0.0.1:34249 - 24113 "TXT IN _redirect._.info.torproxy.git.example.test. udp 60 false 512" NXDOMAIN qr,aa,rd 141 0.000245351s
2019-09-30T22:19:24.774+03:30 [INFO] 127.0.0.1:46892 - 57040 "TXT IN _redirect._._.torproxy.git.example.test. udp 57 false 512" NXDOMAIN qr,aa,rd 138 0.000276778s
2019-09-30T22:19:24.774+03:30 [INFO] 127.0.0.1:49182 - 12356 "TXT IN _redirect._._._.git.example.test. udp 50 false 512" NXDOMAIN qr,aa,rd 131 0.000249735s
```

Comment (erbesharat) [reply path.go]: Just thought of a better way to handle it in case the current code causes some edge cases.
We can get the index of `/info/refs` and split the path from start to that index.

**Comment by stp-ip** [comment]
I agree. Merging now.
Test failure is me "cleaning up" the tests running. aka we are now trying to run non existent tests. If you wanna look into this as a small task ;)
